api/server/app.cc

```diff
@@ -443,6 +443,18 @@ static int InitCgroup(int multiplier) {
          SrcCgroup() + "/butterfly/cpu.shares") {
         LOG_WARNING_("can't set cgroup priority");
     }
+
+    /* new cgroup use a diferent directory for cpushare and cpu, if not, we need to initilize some files */
+    BASH("! cat " + std::string(SrcCgroup()) + "/cpuset.mems >> /dev/null") {
+	    BASH("echo `cat " + std::string(SrcCgroup()) + "/cpuset.mems` > " +
+		 SrcCgroup() + "/butterfly/cpuset.mems") {
+		    LOG_WARNING_("can't set cgroup cpuset.mems");
+	    }
+	    BASH("echo `cat " + std::string(SrcCgroup()) + "/cpuset.cpus` > " +
+		 SrcCgroup() + "/butterfly/cpuset.cpus") {
+		    LOG_WARNING_("can't set cgroup cpuset.cpus");
+	    }
+    }
```

Comment:
- need title fix
- using a tmp variable for `SrcCgroup()` will make it more readable.
- you can put the first string inside a std::string (`echo cat`)
- no warning if the first BASH fails ? Is the first BASH really useful ?